src/block.rs

```diff
@@ -1,33 +1,314 @@
 //use std::stream::Stream;
 //use futures::io::AsyncRead;
 
-#[derive(Debug)]
+use std::mem::transmute;
+use serde::{Serialize, Deserialize};
+
+use data_encoding::HEXLOWER;
+
+use crate::burnfee::BurnFee;
+use crate::crypto::{PublicKey, hash};
+use crate::transaction::Transaction;
+
+use crate::helper::create_timestamp;
+/// Our Block object
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
 pub struct Block {
-    
+    pub body:     BlockBody,
+    pub is_valid: u8,
+    mintid:   u32,
+    maxtid:   u32,
+    bsh:      [u8; 32],
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
+pub struct BlockBody {
+    pub id:          u32,
+    pub ts:          u64,
+    pub prevbsh:     [u8; 32],
+    pub creator:     PublicKey,
+    pub txs:         Vec<Transaction>,
+    pub bf:	         BurnFee,
+    merkle:          [u8; 32],
+    difficulty:      f32,
+    paysplit:        f32,
+    vote:            i8,
+    treasury:        u64,
+    coinbase:        u64,
+    reclaimed:       u64
+}
+
+impl BlockBody {
+    pub fn new(block_creator: PublicKey, prevbsh: [u8;32]) -> BlockBody {
+        return BlockBody {
+    	    id:          0,
+    	    ts:          create_timestamp(),
+    	    prevbsh:     prevbsh,
+    	    merkle:      [0; 32],
+    	    creator:     block_creator,
+    	    txs:         vec![],
+	        bf:          BurnFee::new(0.0, 0),
```

Comment:
Automatic code-formatting would be greta for the whole project. Otherwise i.e. people might be confused if this pupose.